<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>503</width>
    <height>345</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <property name="sizeConstraint">
     <enum>QLayout::SetDefaultConstraint</enum>
    </property>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayoutMain">
      <item>
       <widget class="QFrame" name="frame">
        <property name="sizePolicy">
         <sizepolicy hsizetype="MinimumExpanding" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>200</width>
          <height>200</height>
         </size>
        </property>
        <property name="frameShape">
         <enum>QFrame::StyledPanel</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Raised</enum>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_3">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayoutFrame">
           <item>
            <spacer name="spinnerLeftSpacer">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <spacer name="spinnerRightSpacer">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QFrame" name="settingsFrame">
        <property name="frameShape">
         <enum>QFrame::StyledPanel</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Raised</enum>
        </property>
        <layout class="QGridLayout" name="gridLayout_5">
         <item row="3" column="1">
          <widget class="QDoubleSpinBox" name="fadeSpinBox">
           <property name="value">
            <double>70.000000000000000</double>
           </property>
          </widget>
         </item>
         <item row="5" column="1">
          <widget class="QSpinBox" name="lineLengthSpinBox">
           <property name="value">
            <number>10</number>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="opacityLabel">
           <property name="text">
            <string>Opacity:</string>
           </property>
          </widget>
         </item>
         <item row="4" column="0">
          <widget class="QLabel" name="linesLabel">
           <property name="text">
            <string>Lines:</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QDoubleSpinBox" name="roundnessSpinBox">
           <property name="value">
            <double>70.000000000000000</double>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="fadeLabel">
           <property name="text">
            <string>Fade Perc:</string>
           </property>
          </widget>
         </item>
         <item row="6" column="1">
          <widget class="QSpinBox" name="lineWidthSpinBox">
           <property name="value">
            <number>5</number>
           </property>
          </widget>
         </item>
         <item row="6" column="0">
          <widget class="QLabel" name="lineWidthLabel">
           <property name="text">
            <string>Line Width:</string>
           </property>
          </widget>
         </item>
         <item row="4" column="1">
          <widget class="QSpinBox" name="linesSpinBox">
           <property name="value">
            <number>12</number>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="roundnessLabel">
           <property name="text">
            <string>Roundness:</string>
           </property>
          </widget>
         </item>
         <item row="9" column="1">
          <spacer name="settingsVerticalSpacer">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>40</height>
            </size>
           </property>
          </spacer>
         </item>
         <item row="8" column="0">
          <widget class="QLabel" name="revsLabel">
           <property name="text">
            <string>Rev/S:</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QDoubleSpinBox" name="opacitySpinBox">
           <property name="value">
            <double>15.000000000000000</double>
           </property>
          </widget>
         </item>
         <item row="8" column="1">
          <widget class="QSpinBox" name="revsSpinBox">
           <property name="value">
            <number>1</number>
           </property>
          </widget>
         </item>
         <item row="5" column="0">
          <widget class="QLabel" name="lineLengthLabel">
           <property name="text">
            <string>Line Length:</string>
           </property>
          </widget>
         </item>
         <item row="7" column="0">
          <widget class="QLabel" name="radiusLabel">
           <property name="text">
            <string>Inner Radius:</string>
           </property>
          </widget>
         </item>
         <item row="7" column="1">
          <widget class="QSpinBox" name="radiusSpinBox">
           <property name="value">
            <number>10</number>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <spacer name="verticalSpacer">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>17</width>
        <height>0</height>
       </size>
      </property>
     </spacer>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <spacer name="buttonsSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QPushButton" name="startButton">
        <property name="text">
         <string>Start</string>
        </property>
        <property name="default">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="stopButton">
        <property name="text">
         <string>Stop</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="launchBlockingButton">
        <property name="text">
         <string>Launch 5 Second Blocking</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="pickColourButton">
        <property name="text">
         <string>Pick Colour</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>roundnessSpinBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>setRoundness(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>512</x>
     <y>32</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>6</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>opacitySpinBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>setMinimumTrailOpacity(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>533</x>
     <y>68</y>
    </hint>
    <hint type="destinationlabel">
     <x>24</x>
     <y>4</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>fadeSpinBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>setTrailFadePercentage(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>503</x>
     <y>105</y>
    </hint>
    <hint type="destinationlabel">
     <x>56</x>
     <y>9</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>linesSpinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>setNumberOfLines(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>495</x>
     <y>127</y>
    </hint>
    <hint type="destinationlabel">
     <x>76</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>lineLengthSpinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>setLineLength(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>548</x>
     <y>162</y>
    </hint>
    <hint type="destinationlabel">
     <x>95</x>
     <y>-9</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>lineWidthSpinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>setLineWidth(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>512</x>
     <y>198</y>
    </hint>
    <hint type="destinationlabel">
     <x>5</x>
     <y>32</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>radiusSpinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>setInnerRadius(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>531</x>
     <y>232</y>
    </hint>
    <hint type="destinationlabel">
     <x>113</x>
     <y>-5</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>revsSpinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>setRevolutionsPerSecond(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>496</x>
     <y>266</y>
    </hint>
    <hint type="destinationlabel">
     <x>54</x>
     <y>302</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pickColourButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>showColourDialog()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>495</x>
     <y>324</y>
    </hint>
    <hint type="destinationlabel">
     <x>15</x>
     <y>331</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>launchBlockingButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>launchBlockingSpinner()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>325</x>
     <y>319</y>
    </hint>
    <hint type="destinationlabel">
     <x>76</x>
     <y>309</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>setRoundness(double)</slot>
  <slot>setMinimumTrailOpacity(double)</slot>
  <slot>setTrailFadePercentage(double)</slot>
  <slot>setRevolutionsPerSecond(int)</slot>
  <slot>setNumberOfLines(int)</slot>
  <slot>setLineLength(int)</slot>
  <slot>setLineWidth(int)</slot>
  <slot>setInnerRadius(int)</slot>
  <slot>start()</slot>
  <slot>stop()</slot>
  <slot>showColourDialog()</slot>
  <slot>launchBlockingSpinner()</slot>
 </slots>
</ui>
